src/sds.c

```diff
@@ -124,8 +130,7 @@ sds _sdsnewlen(const void *init, size_t initlen, int trymalloc) {
     s = (char*)sh+hdrlen;
     fp = ((unsigned char*)s)-1;
     usable = usable-hdrlen-1;
-    if (usable > sdsTypeMaxSize(type))
-        usable = sdsTypeMaxSize(type);
+    assert(usable <= sdsTypeMaxSize(type));
```

Comment:
You can't do such assertion here, usable is got from allocator and allocator didn't have such kind of guarantee that it will have maximum size. For example, if I requested 236 bytes, the allocator size could be 248 bytes (good) but maybe 264 bytes. I think the guarantee is that the allocated size >= requested size.

Reply:
You're right, the allocator doesn't guarantee a maximum size. But it doesn't give random blocks either. Jemalloc allocates buffers of pre-defined sizes. And SDS types are aligned accordingly. If this assertion is false, `usable` is incorrect.

Reply:
I thought about jemalloc. I was sure that libc allocator would return a buffer of the exact size. Apparently I was wrong, thank you for noting it. I'll fix it.